OneCommunityGlobal/HighestGoodNetworkApp
# Carlos anniversary celebrated

# Description
This part of a new Report Page: Create Company Summary Dashboard for Admins requested by Jae.
Implements # (WBS)
Anniversary Celebrared Component

## Related PRS (if any):

This frontend PR is related to the [#1128](https://github.com/OneCommunityGlobal/HGNRest/pull/1128) backend PR.


![Screenshot 2024-10-10 091351](https://github.com/user-attachments/assets/ab30f3d7-d93e-4bed-ad2b-e1d49d9bd0f2)

![Screenshot 2024-10-10 091516](https://github.com/user-attachments/assets/3550ebb3-cfd4-4c6f-a922-ede5e70a5f6e)

![Screenshot 2024-10-10 091617](https://github.com/user-attachments/assets/954102f2-7b4a-40e2-9bc0-c85e21a7eb11)

…

## Main features  explained:
- Visualization of volunteer anniversaries within the current week.
- Comparison of volunteer anniversaries between the current and previous week.
- Clicking on the volunteer email icon allows you to send a personalized congratulatory email using previously configured Gmail OAuth settings. (For more information on Gmail setup, please watch the video below.)

…

## How to test:

This frontend PR is related to the [#1128](https://github.com/OneCommunityGlobal/HGNRest/pull/1128) backend PR.

1. Check out the current branch.
2. Do `npm install` and `...` to run this PR locally.
3. Clear site data/cache.
4. Log in  as admin user.
5. Navigate to Dashboard > Reports > Total Org Summary > Volunteer Engagement Trends.
6. Set up the  Gmail Api in your account  http://console.cloud.google.com and https://developers.google.com/  (please watch   the Gmail Api setup video belowuntil minut 7).
7.Set up the environment variables in the .env file of your backend project (see .env image).
8. Chech the anniversary list.
9. Select and click on the email icon.
10. Send a congratulatory email.
11. Check sent folder of your gmail acount.

## Screenshots or videos needed:

### App features video:

https://www.loom.com/share/d4f75956df7b4ef6946dd17aa8bfa806?sid=5ccabce1-1fb8-4c11-a9f7-a5127a3ea793

### Gmail Api setup video (just watch until minut 7)

https://www.youtube.com/watch?v=-rcRf7yswfM&list=LL&index=8&t=304s

### Image of Environment variables

![Screenshot 2024-10-10 093847](https://github.com/user-attachments/assets/0389807a-2a3c-494c-9546-8c723eb50f66)

## Note:
Remember!. Here are  the absolutely necessary steps to successfully send a congratulatory email from your Google account:

- Set up the Gmail API in your account. (Removes unnecessary "Api") - Links can stay as provided. (Please watch the Gmail API setup video  until minute 7).
- Set up the environment variables in the .env file of your backend project. (See  image)
